### PR TITLE
HWINFO: Support NXP LPC family

### PR DESCRIPTION
**Pull request by mmahadevan108**
Add hwinfo driver support for NXP's LPC family.

### CODE REVIEW

**Comment by alexanderwachter** [comment]
Hi @mmahadevan108!
Did you test it on HW? I had an implementation once, but dropped it due to failing test on HW.

**Comment by mmahadevan108** [comment]
Yes, tested it on the MXRT685 board.

**Review by alexanderwachter — APPROVED** [review]
LGTM. Thanks!

File: drivers/hwinfo/hwinfo_mcux_syscon.c [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ssize_t z_impl_hwinfo_get_device_id(uint8_t *buffer, size_t length)
+{
+	uint32_t id[4];
+	uint32_t *idp = id;
+
+	*idp++ = sys_cpu_to_be32(SYSCTL0->UUID[0]);
+	*idp++ = sys_cpu_to_be32(SYSCTL0->UUID[1]);
+	*idp++ = sys_cpu_to_be32(SYSCTL0->UUID[2]);
+	*idp++ = sys_cpu_to_be32(SYSCTL0->UUID[3]);
+
```
Comment (galak) [inline drivers/hwinfo/hwinfo_mcux_syscon.c:20]: Any reason for the pointer and not just doing:

```
id[0] = ...
id[1] = ...
id[2] = ...
id[3] = ...
```

Comment (mmahadevan108) [reply drivers/hwinfo/hwinfo_mcux_syscon.c:19]: Thank you for pointing this out, I have changed it.

File: drivers/hwinfo/hwinfo_mcux_syscon.c [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ * Copyright (c) 2021 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <drivers/hwinfo.h>
+#include <string.h>
+#include <sys/byteorder.h>
+
+ssize_t z_impl_hwinfo_get_device_id(uint8_t *buffer, size_t length)
```
Comment (galak) [inline drivers/hwinfo/hwinfo_mcux_syscon.c:11]: Would like to see the address of SYSCTL0 to get drive by devicetree, but will not hold this up.

Comment (mmahadevan108) [reply drivers/hwinfo/hwinfo_mcux_syscon.c:11]: Thanks, I have updated the implementation to get this information from the device tree.

Comment (mmahadevan108) [reply drivers/hwinfo/hwinfo_mcux_syscon.c:13]: I have also added the hwinfo implementation for the LPC55S6X SoC.

File: dts/arm/nxp/nxp_rt6xx_common.dtsi [hunk 3]
```diff
@@ -66,6 +66,12 @@
 		#clock-cells = <1>;
 	};
 
+	uuid: uuid@2f50 {
+		compatible = "nxp,lpc-uid";
+		reg = <0x2f50 0x10>;
+		label = "HWINFO";
```
Comment (galak) [inline dts/arm/nxp/nxp_rt6xx_common.dtsi:72]: remove label, not needed.

File: dts/arm/nxp/nxp_lpc55S6x_common.dtsi [hunk 4]
```diff
@@ -104,10 +104,16 @@
...
 			compatible = "soc-nv-flash";
-			reg = <0x9D800 DT_SIZE_K(10)>;
+			reg = <0x9D800 DT_SIZE_K(9)>;
 			status = "disabled";
 		};
 
+		uuid: flash@9FC70 {
+			compatible = "nxp,lpc-uid";
+			reg = <0x9FC70 0x10>;
+			label = "HWINFO";
```
Comment (galak) [inline dts/arm/nxp/nxp_lpc55S6x_common.dtsi:114]: remove label, not needed.

**Review by galak — APPROVED** [review]

File: dts/arm/nxp/nxp_lpc55S6x_common.dtsi (hunk 4 above)
Comment (mmahadevan108) [reply dts/arm/nxp/nxp_lpc55S6x_common.dtsi]: Done

File: dts/arm/nxp/nxp_rt6xx_common.dtsi (hunk 3 above)
Comment (mmahadevan108) [reply dts/arm/nxp/nxp_rt6xx_common.dtsi]: Done.

**Review by MaureenHelm — APPROVED** [review]

**Review by galak — APPROVED** [review]

**Comment by galak** [comment]
@mmahadevan108 did you try this on anything other than RT6xx?

**Pull request merged**